inbm/dispatcher-agent/dispatcher/fota/guid.py

```diff
@@ -28,7 +27,7 @@ def extract_guids(fw_tool: str, types: list[str]) -> list[str]:
         raise FotaError("Firmware Update Aborted: failed to list GUIDs: {}".format(str(err)))
     guids = _parse_guids(out, types)
     logger.debug("GUIDs: " + str(guids))
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Compile the types into a regular expression before iterating through the output lines for better performance and readability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    import re
    pattern = re.compile('|'.join([re.escape(type_str) for type_str in types]))
    for line in output.splitlines():
        if pattern.search(line):
            guids.append(line.split(',')[1].split()[0].strip('{').strip('}'))
```